streaming/dataset.py

```diff
@@ -174,98 +174,99 @@ class StreamingDataset(Array, IterableDataset):
     Features elastically deterministic shuffling, which enables fast mid-epoch resumption.
 
     Checkpoints are represented in JSON as follows:
-
-    .. code-block:: json
-
-        {
-            "epoch" :"int",
-            "sample_in_epoch": "int",
-            "shuffle_seed": "int",
-            "num_canonical_nodes": "int"
-        }
-
-    StreamingDataset init takes two kinds of arguments:
-
-    * What to iterate:
-
-      * One or more streams (you must provide either ``streams`` or ``remote``/``local``):
-
-        * ``streams``
-        * ``remote``
-        * ``local``
-
-      * Knobs to control streaming behavior, which, if multiple streams are provided,
-        become defaults applied to each of them:
-
-        * ``split``
-        * ``download_retry``
-        * ``download_timeout``
-        * ``validate_hash``
-        * ``keep_zip``
-
-      * Absolute dataset size, if streams were weighted relatively:
-
-        * ``epoch_size``
-
-    * How to iterate:
-
-      * Shard lifecycle:
-
-        * ``predownload``
-        * ``cache_limit``
-
-      * Sampling:
-
-        * ``sampling_method``
-        * ``sampling_granularity``
-
-      * Determinism:
-
-        * ``partition_algo``
-        * ``num_canonical_nodes``
-        * ``batch_size``
-
-      * Shuffling:
-
-        * ``shuffle``
-        * ``shuffle_algo``
-        * ``shuffle_seed``
-        * ``shuffle_block_size``
-
-      * Batching:
-
-        * ``batching_method``
-
+    >>> from typing import TypedDict
+    >>>
+    >>> class StateDict(TypedDict):
+    >>>     epoch: int
+    >>>     sample_in_epoch: int
+    >>>     shuffle_seed: int
+    >>>     num_canonical_nodes: int
+    >>>     num_physical_nodes: int
+
+    Organization of arguments:
+      * What to iterate:
+          * The streams (either provide streams, or remote and/or local for an implicit stream):
+              * ``epoch_size``
+              * ``streams``
+              * ``remote``
+              * ``local``
+          * Stream arguments (used as defaults if explicit streams, or as args if implicit stream):
+              * Filesystem:
+                  * ``split``
+              * Schema:
+                  * ``allow_schema_mismatch``
+                  * ``allow_unsafe_types``
+                  * ``allow_unchecked_resumption``
+              * Downloads:
+                  * ``download_retry``
+                  * ``download_timeout``
+                  * ``download_max_size``
+                  * ``validate_hash``
+                  * ``keep_phases``
+      * How to iterate:
+          * Init:
+              * ``index_download_procs``
```

Comment:
nit: I think you forgot to remove `index_download_procs`, `index_download_procs_per_cpu`, and `index_download_max_procs`, from here.

Reply:
Fixed thx